### ISSUE TITLE
Underfitting Issues

### ISSUE THREAD

**Issue by dano000** [zylo117/Yet-Another-EfficientDet-Pytorch#440]
Hi there,

So I managed to get the D5 version trained successfully, using the following anchor scales and ratios.

anchor_scales = [2 ** 0, 2 ** (1.0 / 8.0), 2 ** (2.0 / 8.0), 2 ** (3.0 / 8.0), 2 ** (4.0 / 8.0), 2 ** (5.0 / 8.0), 2 ** (6.0 / 8.0), 2 ** (7.0 / 8.0)]
anchor_ratios = [(1.2, 0.8), (1.5, 0.7), (1.9, 0.5), (2.6, 0.4), (3.5, 0.3), (4.5, 0.2), (5.2, 0.2), (5.6, 0.2), (6.3, 0.2), (7.4, 0.1), (8.2, 0.1), (9.2, 0.1)]

I used this tool to calculate the anchor ratios: https://github.com/mnslarcher/kmeans-anchors-ratios

and manually entered those anchor scales, so they haven't been calculated or optimised. 

This result was after approx 261 epochs, 312000 steps, approx 45 hours of training.

However the model seems to be under fitting quite a lot. It doesn't seem to work well on most validation data. It maxes out at about 0.75 threshold for all classes, _and it doesn't seem to perform very well._ 

Currently I get better results with ResNet 101 from detectron2.

Should I reduce the number of anchor_ratios and anchor scales?

This was on a black and white data-set, so the mean / std of the data-set was adjusted to:

mean: [0.9646505, 0.9646505, 0.9646505]
std: [0.17946596, 0.17946596, 0.17946596]

Does anyone have any tips on improving performance and increasing accuracy?

Should I reduce the amount of anchor ratios I have? Increase the amount of anchor ratios? Change the anchor scales? Find the mean and standard dev. of the colours of my data-set differently?

Any help would be greatly appreciated

Thanks!

**Comment by zylo117** [comment]
That's too many scales and ratios, for most cases, 3x3 is enough.
Why did you set `mean` to 0.96? That's almost 1, are all of your images are binary, either 0 or 255? I think mean and std should be set to 0.5.

**Comment by dano000** [comment]
Ok I'll try with less.

Yes the mean / std values are correct, since the majority not the dataset
was white pixels, so this is to be expected.

I'll let you know how it goes.

Thanks.

**Comment by LoveIsAGame** [comment]
The main reason is that reg stopped decreasing a long time ago, about the 1000th epoch
